src/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -3399,18 +3399,47 @@ public double getLiveTombstoneRatio()
 
     private double getDroppableTombstoneRatio(boolean useGcGrace)
     {
-        double allDroppable = 0;
         long allColumns = 0;
-        int localTime = (int)(System.currentTimeMillis()/1000);
+        Collection<SSTableReader> sstables = getSSTables();
+        for (SSTableReader sstable : sstables)
+        {
+            allColumns += sstable.getEstimatedColumnCount().mean() * sstable.getEstimatedColumnCount().count();
+        }
+        double allDroppable = getDroppableTombstoneCount(sstables, useGcGrace);
+        return allColumns > 0 ? allDroppable / allColumns : 0;
+    }
 
-        for (SSTableReader sstable : getSSTables())
+    public double getDroppableTombstoneCount()
+    {
+        return getDroppableTombstoneCount(true);
+    }
+
+    public double getTombstoneCount()
+    {
+        return getDroppableTombstoneCount(false);
+    }
+
+    public double getLiveTombstoneCount()
+    {
+        return getTombstoneCount() - getDroppableTombstoneCount();
+    }
+
+    private double getDroppableTombstoneCount(boolean useGcGrace)
+    {
+        return getDroppableTombstoneCount(getSSTables(), useGcGrace);
+    }
+
+    private double getDroppableTombstoneCount(Collection<SSTableReader> sstables, boolean useGcGrace) {
```

Comment:
nit to match c* style
```suggestion
    private double getDroppableTombstoneCount(Collection<SSTableReader> sstables, boolean useGcGrace)
    {
```